akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedStashManagement.scala

```diff
@@ -0,0 +1,61 @@
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.{ DeadLetter, StashOverflowException }
+import akka.actor.typed.scaladsl.{ ActorContext, StashBuffer }
+import akka.event.{ Logging, LoggingAdapter }
+import akka.persistence._
+import akka.util.ConstantFun
+import akka.{ actor ⇒ a }
+
+trait EventsourcedStashManagement {
+  import akka.actor.typed.scaladsl.adapter._
+
+  protected def log: LoggingAdapter
+
+  protected def extension: Persistence
+  protected def context: ActorContext[Any]
+
+  protected val internalStash: StashBuffer[Any]
+
+  /**
+   * The returned [[StashOverflowStrategy]] object determines how to handle the message failed to stash
+   * when the internal Stash capacity exceeded.
+   */
+  protected val internalStashOverflowStrategy: StashOverflowStrategy =
+    extension.defaultInternalStashOverflowStrategy match {
+      case ReplyToStrategy(_) ⇒
+        throw new RuntimeException("ReplyToStrategy is not supported in Akka Typed, since there is no sender()!")
+      case other ⇒
+        other // the other strategies are supported
+    }
+
+  protected def stash(msg: Any, behavior: Behavior[Any]): Behavior[Any] = {
```

Comment:
the `behavior` isn't used here for anything else than return value, better to have Unit and let the caller return the behavior

Reply:
Good simplification, done

Reply:
Makes sense to only return behaviour where it *has to* be the handled one